src/datasets/packaged_modules/spark/spark.py

```diff
@@ -196,6 +225,11 @@ def write_arrow(it):
                     names=["task_id", "num_examples", "num_bytes"],
                 )
 
+            if working_fpath != fpath:
+                for file in os.listdir(os.path.dirname(working_fpath)):
+                    dest = os.path.join(os.path.dirname(fpath), os.path.basename(file))
+                    os.rename(file, dest)
```

Comment:
os.rename doesn't always work depending on the source/target disks, you can use shutil.move instead

Reply:
Thanks, done!